server/src/main/java/io/unitycatalog/server/security/SecurityContext.java

```diff
@@ -0,0 +1,126 @@
+package io.unitycatalog.server.security;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.unitycatalog.server.UnityCatalogServer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SecurityContext {
+
+  public interface Issuers {
+    String INTERNAL = "internal";
+  }
+
+  private static final ObjectMapper mapper = new ObjectMapper();
+  private static final Logger LOGGER = LoggerFactory.getLogger(UnityCatalogServer.class);
+
+  private final Path certsFile;
+  private final Path serviceTokenFile;
+
+  @Getter private final RSAPublicKey rsaPublicKey;
+  @Getter private final RSAPrivateKey rsaPrivateKey;
+  @Getter private final Algorithm algorithm;
+  @Getter private final String serviceToken;
+  @Getter private final String keyId;
+  @Getter private final String serviceName;
+  @Getter private final String localIssuer;
+
+  @SneakyThrows
+  public SecurityContext(
+      Path configurationFolder,
+      SecurityConfiguration securityConfiguration,
+      String serviceName,
+      String localIssuer) {
+    this.serviceName = serviceName;
+    this.localIssuer = localIssuer;
+
+    certsFile = configurationFolder.resolve("certs.json");
+    serviceTokenFile = configurationFolder.resolve("token.txt");
+
+    rsaPublicKey = securityConfiguration.rsaPublicKey();
+    rsaPrivateKey = securityConfiguration.rsaPrivateKey();
+    algorithm = securityConfiguration.algorithmRSA();
+    keyId = securityConfiguration.getKeyId();
+
+    // Do this every time because the SecurityConfiguration parameters
+    // might have changed since last started.
+    serviceToken = createServiceToken();
+
+    createInternalCertsFile();
+    createServiceTokenFile();
+
+    LOGGER.info("--- Internal Certs Configuration --");
+    LOGGER.info(getInternalCertsFile());
+  }
+
+  public String createAccessToken(DecodedJWT decodedJWT) {
+    return JWT.create()
+        .withSubject(serviceName)
+        .withIssuer(localIssuer)
+        .withIssuedAt(new Date())
+        .withKeyId(keyId)
+        .withJWTId(UUID.randomUUID().toString())
+        .withClaim(JwtClaim.TOKEN_TYPE.key(), JwtTokenType.ACCESS.name())
+        .withClaim(JwtClaim.SUBJECT.key(), decodedJWT.getClaim("email").asString())
+        .sign(algorithm);
+  }
+
+  public String createServiceToken() {
```

Comment:
The service token creation is not exposed via any API -- is this for the future to support admin access tokens?

Reply:
Yeah this is mainly for the future. In other platforms, we had different types of tokens that are used in specific contexts so I wanted to build that here, although for now there is no practical difference between service tokens and (user) access tokens.